circuits/src/transaction.rs

```diff
@@ -395,33 +398,43 @@ impl<const H: usize, const I: usize> TxCircuit<H, I> {
     pub fn new(
         tx_input_notes: [TxInputNote<H>; I],
         tx_output_notes: [TxOutputNote; TX_OUTPUT_NOTES],
-        skeleton_hash: BlsScalar,
+        payload_hash: BlsScalar,
         root: BlsScalar,
         deposit: u64,
         max_fee: u64,
+        rp: RecipientParameters,
     ) -> Self {
         Self {
             tx_input_notes,
             tx_output_notes,
-            skeleton_hash,
+            payload_hash,
             root,
             deposit,
             max_fee,
+            rp,
         }
     }
 }
 
 impl<const H: usize, const I: usize> Circuit for TxCircuit<H, I> {
     fn circuit(&self, composer: &mut Composer) -> Result<(), Error> {
-        gadget::<H, I>(
+        // Make the payload hash a public input of the circuit
+        let payload_hash = composer.append_public(self.payload_hash);
+
+        // Nullify all the tx input notes
+        nullify_gadget::<H, I>(
```

Comment:
Why is this gadget called nullify..?

Reply:
to be a bit more self-explanatory within the crate. Do you have another proposal? tx_gadget? maybe?

Reply:
considering it is the whole transfer circuit, how about `transfer_gadget`?